lib/ansible_tower_client/base_models/job_template.rb

```diff
@@ -25,5 +37,20 @@ def extra_vars_hash
     def override_raw_attributes
       { :credential => :credential_id, :inventory => :inventory_id, :project => :project_id }
     end
+
+    private
+
+    def validate_launch_options(options)
+      ignored_options = REQUIRED_ATTRIBUTES_FOR_OPTIONS.select do |option, checkmark|
+        options.slice(option.to_sym, option.to_s).any?(&:present?) && !(respond_to?(checkmark) && send(checkmark))
+      end.keys
+
+      return if ignored_options.empty?
+
+      message = ignored_options.map(&:inspect).join(', ')
+      message = ignored_options.size == 1 ? " #{message} is" : "s #{message} are"
+      message = "Option#{message} provided but corresponding ask on launch flag has not been turn on"
```

Comment:
The message needs some rephrase. I initially gave you some idea but not meant to be final words.

Reply:
Looks like my other comment got collapsed https://github.com/ansible/ansible_tower_client_ruby/pull/119#discussion_r213082218

Reply:
No I saw that. And that your proposal looks very ugly. However, if you insist...
But it will not be my code, it will be your code.

Reply:
@bzwei Please tell me what exactly you want here. I'm not a native English speaking person. I'm not a science fiction writer, or anything else like this.
Please don't ask me to "go somewhere I don't know where, and find something I don't know what".

Reply:
@AlexanderZagaynov Breaking up the message like this makes it difficult if we ever need to translate the string.

I prefer that we use a simple message that lists the flags as the end:
```ruby
raise ArgumentError, "'PROMPT ON LAUNCH' is required for the following fields: #{ignored_options.join(', ')}"
```
I removed the `map(&:inspect)` logic as well since I do not feel it is needed.


Note I am using the wording from the Tower UI for the checkbox name:
![image](https://user-images.githubusercontent.com/2591569/45446619-6413b780-b69b-11e8-9e99-4db70ce0216b.png)

Reply:
Hi @gmcculloug! Thank you for such valuable comment!
It's fixed now.